tutorials/llm/llama/domain-adaptive-pretraining/code/custom_tokenization.ipynb

```diff
@@ -0,0 +1,2569 @@
+      "  from .autonotebook import tqdm as notebook_tqdm\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "import torch\n",
+    "from datasets import Dataset\n",
+    "from datasets import IterableDataset\n",
+    "from datasets import load_dataset\n",
+    "import jsonlines\n",
+    "import glob\n",
+    "from tokenizers import (\n",
+    "    decoders,\n",
+    "    models,\n",
+    "    normalizers,\n",
+    "    pre_tokenizers,\n",
+    "    processors,\n",
+    "    trainers,\n",
+    "    Tokenizer,\n",
+    ")\n",
+    "from transformers import AutoTokenizer\n",
+    "from tokenization_helper import *\n",
+    "from extend_tokenizer_utils import extend_tokenizer, extend_tokenizer_high_freq_tokens\n",
+    "from get_high_freq_tokens import get_high_freq_tokens\n",
+    "from util import load_weights, merge_embed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "24d00d02",
+   "metadata": {},
+   "source": [
+    "## Step 1: Download llama-2-70b embedding model and tokenizer (Original Tokenizer). Convert the orginal weights to trainable format and save. \n",
+    "\n",
+    "The Original Tokenizer model used here is the llama2 tokenizer which is a Byte Pair Encoding (BPE) model based on sentencepiece."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "7f0c3988",
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Snapshot 0\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n",
+      "Snapshot 1\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n",
+      "Snapshot 2\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n",
+      "Snapshot 3\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n",
+      "Snapshot 4\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n",
+      "Snapshot 5\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n",
+      "Snapshot 6\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n",
+      "Snapshot 7\n",
+      "Embedding layer dimension:  torch.Size([32000, 1024])\n",
+      "Output layer dimension:  torch.Size([4000, 8192])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# download llam2-70b model weights and tokenizer \n",
+    "# ! huggingface-cli download meta-llama/Llama-2-70b --local-dir ./models/weight/llama2-hf/\n",
```

Comment:
Uncomment this line to actually download

Reply:
done